### PR TITLE
Add sincosf function

### LINKED ISSUE #7260: Missing symbol in compiler_rt in release modes
When compiling [Zig Showdown](https://github.com/zig-community/Zig-Showdown) for release, i get the following error when not linking libc/libm: 

```
ld.lld: error: undefined symbol: sincosf
```

The symbol sincosf is not used in any file in the project, so i assume it is required to be in `compiler_rt` as a compiler optimization kicks in

### PR DESCRIPTION
**Pull request by daurnimator**
Fixes #7260

@MasterQ32 I have no idea how to test this, you didn't leave a reproducer.

### CODE REVIEW

**Comment by MasterQ32** [comment]
> @MasterQ32 I have no idea how to test this, you didn't leave a reproducer.

Me neither. It just happens somewhere in > 5000 LOC where code uses a lot of sin and cos

File: lib/std/special/c.zig [hunk 1]
```diff
@@ -634,6 +634,11 @@ export fn cosf(a: f32) f32 {
     return math.cos(a);
 }
 
+export fn sincosf(a: f32, r_sin: *f32, r_cos: *f32) void {
+    r_sin.* = math.sin(a);
+    r_sin.* = math.cos(a);
```
Comment (LemonBoy) [inline lib/std/special/c.zig:639]:
```suggestion
    r_cos.* = math.cos(a);
```

**Comment by LemonBoy** [comment]
[Here's](https://www.man7.org/linux/man-pages/man3/sincosf.3.html) the man page, some tests are needed to make sure this implementation matches the libc one (wrt nan/infinity inputs, the errno part should be honored too?).

Adding the `sincos` variant is trivial.

File: lib/std/special/c.zig (hunk 1 above)
Comment (daurnimator) [reply lib/std/special/c.zig]: doh.

**Comment by daurnimator** [comment]
> the errno part should be honored too?

great question. At first I thought so, but then I saw that e.g. `sinf` doesn't....
What *is* the correct behaviour here?
The fact that LLVM optimises to things that require `errno` doesn't sit well with me.

**Comment by LemonBoy** [comment]
> The fact that LLVM optimises to things that require errno doesn't sit well with me.

Calm your tits, LLVM is smart. The optimization kicks in iff `sin`/`cos` are marked with `readnone` meaning they don't touch `errno` at all. 
Check out this [gcc](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=80042) ticket for more info, the `errno` clause is specified in the sacred [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sin.html) texts.

**Comment by LemonBoy** [comment]
Reproducer:
```zig
const std = @import("std");

extern var foo: f32;

pub fn main() !void {
    var y = @sin(foo);
    var z = @cos(foo);
    std.mem.doNotOptimizeAway(y);
    std.mem.doNotOptimizeAway(z);
}
```

Compile with: `zig build-exe -target native-linux-gnu -OReleaseFast`

**Comment by daurnimator** [comment]
Tested with
```zig
const std = @import("std");

export fn myfunc(foo: f32) f32 {
    var y = @sin(foo);
    var z = @cos(foo);
    return y + z;
}
```
```
zig build-lib -dynamic -target native-linux-gnu -OReleaseFast sincosf.zig
objdump --disassemble libsincosf.so
```

and looks correct.... will re-kick CI because error message is gone now.

**Pull request merged**